roxie/ccd/ccdstate.cpp

```diff
@@ -2588,12 +2588,30 @@ class CRoxiePackageSetManager : implements IRoxieQueryPackageManagerSet, impleme
             }
             else if (strieq(queryName, "control:memlock"))
             {
-                roxiemem::lockRoxieMem(true);
+                int srtn = roxiemem::lockRoxieMem(true);
+                if (!srtn)
+                {
+                    topology->setPropBool("@heapLockMemory", true);
+                    reply.append(" <heapLockMemory>1</heapLockMemory>\n");
```

Comment:
I'd prefer we were consistent with the xml formatting.  In the successful replies the lock state is the content, but in the unsuccessful replies, the lock state is an attribute.

So for consistency, 

`reply.append(" <heapLockMemory locked='1'/>\n");`

would be better.